x-pack/plugin/logstash/src/test/java/org/elasticsearch/xpack/logstash/action/TransportGetPipelineActionTests.java

```diff
@@ -38,6 +49,7 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
```

Comment:
Is there something breaking these tests on thread leak detection now?

Reply:
Couldn't figure out why my test case `testGetPipelinesByExplicitAndWildcardIds` is causing thread leak. Do you have any?

```
// log
com.carrotsearch.randomizedtesting.ThreadLeakError: 1 thread leaked from SUITE scope at org.elasticsearch.xpack.logstash.action.TransportGetPipelineActionTests: 
   1) Thread[id=24, name=elasticsearch[testGetPipelinesByExplicitAndWildcardIds][[timer]], state=TIMED_WAITING, group=TGRP-TransportGetPipelineActionTests]
        at java.base@17.0.2/java.lang.Thread.sleep(Native Method)
        at app//org.elasticsearch.threadpool.ThreadPool$CachedTimeThread.run(ThreadPool.java:671)

	at __randomizedtesting.SeedInfo.seed([DACFC603D5697F8A]:0)
```

Reply:
It looks like there is a ThreadPool instance that hasn't been shut down at the end of the test? Though I'm not seeing one set up at all - perhaps its something downstream of some classes in the test?

Reply:
Ah, found it - it's the NoOpClient - it creates a TestThreadPool instance which is leaving a timer thread laying about at the end of the test. Solution is simple - close the NoOpClient in a try-with-resources block